src/node_worker.cc

```diff
@@ -132,7 +132,14 @@ class WorkerThreadData {
  public:
   explicit WorkerThreadData(Worker* w)
     : w_(w) {
-    CHECK_EQ(uv_loop_init(&loop_), 0);
+    int ret = uv_loop_init(&loop_);
+    if (ret != 0) {
+      char err_buf[128];
+      uv_err_name_r(ret, err_buf, sizeof(err_buf));
+      std::string error_str = SPrintF("ERR_WORKER_INIT_FAILED: %s", err_buf);
+      w->custom_error_ = error_str.c_str();
+      return;
```

Comment:
You may need to add an additional field to this class that tracks whether `uv_loop_init()` succeeded; `CheckedUvLoopClose()` should not be called if `uv_loop_init()` failed.

Reply:
@addaleax, I added a new field `loop_init_failed_` to cater to this.

With this change it was failing on a different assertion in the worker destructor. So I also set `stopped_` to true here. I believe that is the right thing to do.

Reply:
Can you also add a `CHECK(!loop_init_failed_)` before the `uv_run()` call on line 212?

There should probably also be a `CHECK(!data.loop_init_failed_)` after the `if (isolate_ == nullptr) return;` statement in `Worker::Run()`.

`WorkerThreadData` is a friend class of `Worker`. You have access to its private members.

Reply:
@bnoordhuis, Thanks. Addressed your comment in respected lines.